extensions/wikia/PortableInfobox/tests/nodes/NodeDataTest.php

```diff
@@ -39,93 +39,409 @@ public function sourceDataProvider() {
-			[ '<group>' .
-				'<data source="test"><label>labelInsideGroup</label></data>' .
-				'<data source="test2"><label>labelInsideGroup2</label></data>' .
-				'</group>',
-				[], ['test' =>'labelInsideGroup', 'test2' =>'labelInsideGroup2'] ],
-			[ '<group>' .
-				'<title source="title"/>' .
-				'<image source="image"/>' .
-				'<data source="test"><label>labelInsideGroup</label></data>' .
-				'<data source="test2"><label>labelInsideGroup2</label></data>' .
-				'</group>',
-				[], ['test' =>'labelInsideGroup', 'test2' =>'labelInsideGroup2', 'title' => '', 'image' => ''] ],
-			[ '<data source="test"><default>{{{test 2}}}</default></data>', [ ], [ 'test' => '', 'test 2' => '' ] ],
-			[ '<data source="test1"><default>{{#if: {{{test2|}}}| [[{{{test2}}} with some text]] }}</default></data>',
-				[ ], [ 'test1' => '', 'test2' => '' ] ],
-			[ '<data><default>{{#switch: {{{test2|}}}|{{{test3}}}|{{{test4|kdjk|sajdkfj|}}}]] }}</default></data>',
-				[ ], [ 'test2' => '', 'test3' => '', 'test4' => '' ] ],
-			[ '<data source="test1">' .
+			[
+				'<data source="test"></data>',
+				[],
+				[
+					'test' => [
+						'label' => '',
+						'primary' => true
+					]
+				]
+			],
+			[
+				'<image source="test"/>',
+				[],
+				[
+					'test' => [
+						'label' => '',
+						'primary' => true
+					]
+				]
+			],
+			[
+				'<title source="test"/>',
+				[],
+				[
+					'test' => [
+						'label' => '',
+						'primary' => true
+					]
+				]
+			],
+			[
+				'<data source="test"><default>{{{test}}}</default></data>',
+				[],
+				[
+					'test' => [
+						'label' => '',
+						'primary' => true
+					]
+				]
+			],
+			[
+				'<data source="test"><label source="test">{{{test}}}</label><default>{{{test}}}</default></data>',
+				[],
+				[
+					'test' => [
+						'label' => '{{{test}}}',
+						'primary' => true
+					]
+				]
+			],
+			[
+				'<data source="test"><label>testLabel</label></data>',
+				[],
+				[
+					'test' => [
+						'label' => 'testLabel',
+						'primary' => true
+					]
+				]
+			],
+			[ '<data></data>', [], [] ],
+			[
+				'<data source="test"><default>{{{test 2}}}</default></data>',
+				[],
+				[
+					'test' => [
+						'label' => '',
+						'primary' => true
+					],
+					'test 2' => [
+						'label' => ''
+					]
+				]
+			],
+			[
+				'<data source="test1"><default>{{#if: {{{test2|}}}| [[{{{test2}}} with some text]] }}</default></data>',
+				[],
+				[
+					'test1' => [
+						'label' => '',
+						'primary' => true
+					],
+					'test2' => [
+						'label' => ''
+					]
+				]
+			],
+			[
+				'<data><default>{{#switch: {{{test2|}}}|{{{test3}}}|{{{test4|kdjk|sajdkfj|}}}]] }}</default></data>',
+				[],
+				[
+					'test2' => [
+						'label' => ''
+					],
+					'test3' => [
+						'label' => ''
+					],
+					'test4' => [
+						'label' => ''
+					]
+				]
+			],
+			[
+				'<data source="test1">' .
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "<data source="test1">" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)